neurox/interpretation/gaussian_probe.py

```diff
@@ -0,0 +1,282 @@
+        If set to True, actual predictions are also returned along with scores
+        for further use. Defaults to False.
+    metrics : str, optional
+        Metric to use for evaluation scores. For supported metrics see
+        ``interpretation.metrics``
+    Returns
+    -------
+    scores : dict
+        The overall score on the given data with the key ``__OVERALL__``. If
+        ``idx_to_class`` mapping is provided, additional keys representing each
+        class and their associated scores are also part of the dictionary.
+    predictions : list of 3-tuples, optional
+        If ``return_predictions`` is set to True, this list will contain a
+        3-tuple for every input sample, representing
+        ``(source_token, predicted_class, was_predicted_correctly)``
+    """
+    probe.test_features = torch.tensor(X_test).to(probe.device)
+
+    probe.test_labels = torch.tensor(y_test).to(probe.device).long()
+    probe.feature_sets["test"] = probe.test_features
+    probe.label_sets["test"] = probe.test_labels
+
+    probe.test_categorical = probe._get_categorical("test")
+
+    probe.categorical_sets["test"] = probe.test_categorical
+
+    probe._get_distributions(selected_neurons)
+
+    probe._compute_probs(selected_neurons, "test")
+    preds, labels, test_acc, test_mi, test_nmi = probe._predict("test")
+    result = metrics.compute_score(preds, labels, metric)
+
+    if return_predictions:
+        return preds, result
+    return result
+
+
+def get_neuron_ordering(probe, num_of_neurons):
+    """
+    Get global ordering of neurons from a trained probe.
+    This method returns the global ordering of neurons in a model based on
+    the Gaussian Method.
+
+    Parameters
+    ----------
+    probe : interpretation.gaussian_probe.GaussianProbe
+        Trained probe model
+    num_of_neurons: the number of neurons you want, the method requires a not too large number (<400)
+    Returns
+    -------
+    neuron_ordering : list
+        Numpy array of size ``NUM_NEURONS`` with neurons in decreasing order
+        of importance.
+
+    """
+    selected_neurons = []
+    for num_of_neuron in range(0, num_of_neurons):
+        best_neuron = -1
+        best_acc = 0.0
+        best_mi, best_nmi = float("-inf"), float("-inf")
+
+        acc_on_best_mi = 0.0
+        mi_on_best_acc, nmi_on_best_acc = 0.0, 0.0
+        for neuron in range(probe.train_features.shape[1]):
+            if neuron in selected_neurons:
+                continue
+            probe._get_distributions(selected_neurons + [neuron])
+            probe._compute_probs(selected_neurons + [neuron], "train")
+            preds, labels, acc, mi, nmi = probe._predict("train")
+
+            if mi > best_mi:
+                best_mi = mi
+                best_nmi = nmi
+                best_neuron = neuron
+                acc_on_best_mi = acc
+
+        selected_neurons.append(best_neuron)
+        probe._get_distributions(selected_neurons)
```

Comment:
Do we need 279-281? Can we remove the 3 lines (since we are just returning `selected_neurons`